### PR TITLE
KAA-1650: Don't set the body present flag in extension options if body size is equal to zero

### PR DESCRIPTION
**Pull request by RostakaGmfun**
/cc @GamovCoder @sergio-dev

### CODE REVIEW

**Review by garniy — DISMISSED** [review]

File: server/common/server-shared/src/main/java/org/kaaproject/kaa/server/sync/platform/BinaryEncDec.java [hunk 1]
```diff
@@ -424,7 +424,8 @@ private void encode(GrowingByteBuffer buf, LogServerSync logSync) {
   private void encode(GrowingByteBuffer buf, ConfigurationServerSync configurationSync) {
     int option = 0;
     boolean confSchemaPresent = configurationSync.getConfSchemaBody() != null;
-    boolean confBodyPresent = configurationSync.getConfDeltaBody() != null;
+    boolean confBodyPresent = configurationSync.getConfDeltaBody() != null
+        && configurationSync.getConfDeltaBody().array().length != 0;
```
Comment (vtkhir) [inline server/common/server-shared/src/main/java/org/kaaproject/kaa/server/sync/platform/BinaryEncDec.java]: Call [hasArray()](https://docs.oracle.com/javase/8/docs/api/java/nio/ByteBuffer.html#hasArray--) method before invoking [array()](https://docs.oracle.com/javase/8/docs/api/java/nio/ByteBuffer.html#array--)

**Review by vtkhir — CHANGES_REQUESTED** [review]

File: server/common/server-shared/src/main/java/org/kaaproject/kaa/server/sync/platform/BinaryEncDec.java [hunk 2]
```diff
@@ -423,8 +423,12 @@ private void encode(GrowingByteBuffer buf, LogServerSync logSync) {
 
   private void encode(GrowingByteBuffer buf, ConfigurationServerSync configurationSync) {
     int option = 0;
-    boolean confSchemaPresent = configurationSync.getConfSchemaBody() != null;
-    boolean confBodyPresent = configurationSync.getConfDeltaBody() != null;
+    boolean confSchemaPresent = configurationSync.getConfSchemaBody() != null
```
Comment (vtkhir) [inline server/common/server-shared/src/main/java/org/kaaproject/kaa/server/sync/platform/BinaryEncDec.java]: Please extract `configurationSync.getConfSchemaBody()` to local variable

File: server/common/server-shared/src/main/java/org/kaaproject/kaa/server/sync/platform/BinaryEncDec.java [hunk 3]
```diff
@@ -423,8 +423,12 @@ private void encode(GrowingByteBuffer buf, LogServerSync logSync) {
...
+        && configurationSync.getConfSchemaBody().hasArray()
+        && configurationSync.getConfSchemaBody().array().length != 0;
+    boolean confBodyPresent = configurationSync.getConfDeltaBody() != null
```
Comment (vtkhir) [inline server/common/server-shared/src/main/java/org/kaaproject/kaa/server/sync/platform/BinaryEncDec.java]: Please extract `configurationSync.getConfDeltaBody()` to local variable

File: server/common/server-shared/src/main/java/org/kaaproject/kaa/server/sync/platform/BinaryEncDec.java [hunk 4]
```diff
@@ -423,8 +423,12 @@ private void encode(GrowingByteBuffer buf, LogServerSync logSync) {
...
+        && configurationSync.getConfSchemaBody().hasArray()
+        && configurationSync.getConfDeltaBody().array().length != 0;
```
Comment (vtkhir) [inline server/common/server-shared/src/main/java/org/kaaproject/kaa/server/sync/platform/BinaryEncDec.java]: Fix the typo

**Review by vtkhir — CHANGES_REQUESTED** [review]

**Review by vtkhir — APPROVED** [review]

**Comment by rasendubi** [comment]
Jenkins failure is caused by KAA-1640 and is not related to the PR.

**Pull request merged**